File: src/test/fixtures/kotlin/Nesting.kt
// clean — not flagged by nesting
fun cleanShallowNesting(x: Int): Int {
    if (x > 0) {
        if (x > 10) {
            return x
        }
    }
    return 0
}

// flagged — nesting (medium)
fun flaggedDeepNesting(x: Int): Int {
    if (x > 0) {
        if (x > 1) {
            if (x > 2) {
                if (x > 3) {
                    if (x > 4) {
                        return x
                    }
                }
            }
        }
    }
    return 0
}

// flagged — nesting (high)
fun flaggedSevereNesting(x: Int): Int {
    if (x > 0) {
        if (x > 1) {
            if (x > 2) {
                if (x > 3) {
                    if (x > 4) {
                        if (x > 5) {
                            if (x > 6) {
                                return x
                            }
                        }
                    }
                }
            }
        }
    }
    return 0
}

// flagged — nesting
fun flaggedTryNesting(x: Int): Int {
    try {
        try {
            try {
                try {
                    try {
                        return x
                    } catch (e: Exception) {
                        return 0
                    }
                } catch (e: Exception) {
                    return 0
                }
            } catch (e: Exception) {
                return 0
            }
        } catch (e: Exception) {
            return 0
        }
    } catch (e: Exception) {
        return 0
    }
}

File: src/test/fixtures/fsharp/Nesting.fs
module Nesting

// clean — not flagged by nesting
let cleanShallowNesting (x: int) =
    if x > 0 then
        if x > 10 then x else 0
    else
        0

// flagged — nesting (medium)
let flaggedDeepNesting (x: int) =
    if x > 0 then
        if x > 1 then
            if x > 2 then
                if x > 3 then
                    if x > 4 then x else 0
                else
                    0
            else
                0
        else
            0
    else
        0

// flagged — nesting (high)
let flaggedSevereNesting (x: int) =
    if x > 0 then
        if x > 1 then
            if x > 2 then
                if x > 3 then
                    if x > 4 then
                        if x > 5 then
                            if x > 6 then x else 0
                        else
                            0
                    else
                        0
                else
                    0
            else
                0
        else
            0
    else
        0

// flagged — nesting
let flaggedTryNesting (x: int) =
    try
        try
            try
                try
                    try
                        x
                    with _ ->
                        0
                with _ ->
                    0
            with _ ->
                0
        with _ ->
            0
    with _ ->
        0
